java-checks/src/main/java/org/sonar/java/checks/spring/StatusCodesOnResponseCheck.java

```diff
@@ -0,0 +1,222 @@
+    "PRECONDITION_FAILED",
+    "PAYLOAD_TOO_LARGE",
+    "URI_TOO_LONG",
+    "UNSUPPORTED_MEDIA_TYPE",
+    "REQUESTED_RANGE_NOT_SATISFIABLE",
+    "EXPECTATION_FAILED",
+    "I_AM_A_TEAPOT",
+    "UNPROCESSABLE_ENTITY",
+    "LOCKED",
+    "FAILED_DEPENDENCY",
+    "UPGRADE_REQUIRED",
+    "PRECONDITION_REQUIRED",
+    "TOO_MANY_REQUESTS",
+    "REQUEST_HEADER_FIELDS_TOO_LARGE",
+    "UNAVAILABLE_FOR_LEGAL_REASONS",
+    "INTERNAL_SERVER_ERROR",
+    "NOT_IMPLEMENTED",
+    "BAD_GATEWAY",
+    "SERVICE_UNAVAILABLE",
+    "GATEWAY_TIMEOUT",
+    "HTTP_VERSION_NOT_SUPPORTED",
+    "VARIANT_ALSO_NEGOTIATES",
+    "INSUFFICIENT_STORAGE",
+    "LOOP_DETECTED",
+    "BANDWIDTH_LIMIT_EXCEEDED",
+    "NOT_EXTENDED",
+    "NETWORK_AUTHENTICATION_REQUIRED");
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return List.of(Tree.Kind.CLASS);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    ClassTree classTree = (ClassTree) tree;
+    if (!isClassController(classTree)) {
+      return;
+    }
+
+    MethodInvocationVisitor methodInvocationVisitor = new MethodInvocationVisitor();
+    classTree.accept(methodInvocationVisitor);
+  }
+
+  private class MethodInvocationVisitor extends BaseTreeVisitor {
+
+    MethodMatchers statusMethodMatchers = MethodMatchers.create()
+      .ofTypes(RESPONSE_ENTITY)
+      .names("status")
+      .addParametersMatcher("org.springframework.http.HttpStatus")
+      .build();
+
+    MethodMatchers okMethodMatchers = MethodMatchers.create()
+      .ofTypes(RESPONSE_ENTITY)
+      .names("ok", "created", "accepted", "noContent")
+      .withAnyParameters()
+      .build();
+
+    MethodMatchers errorMethodsMatcher = MethodMatchers.create()
+      .ofTypes(RESPONSE_ENTITY)
+      .names("badRequest", "notFound", "unprocessableEntity")
+      .addWithoutParametersMatcher()
+      .build();
+
+    @Override
+    public void visitMethodInvocation(MethodInvocationTree methodInvocationTree) {
+
+      if (statusMethodMatchers.matches(methodInvocationTree)) {
+        checkTryCatch(methodInvocationTree);
+      }
+
+      if (okMethodMatchers.matches(methodInvocationTree)) {
+        Tree catchParent = checkCatch(methodInvocationTree, false);
+        if (catchParent == null) {
+          checkTry(methodInvocationTree, true);
+        }
+      }
+
+      if (errorMethodsMatcher.matches(methodInvocationTree)) {
+        Tree catchParent = checkCatch(methodInvocationTree, true);
+        if (catchParent == null) {
+          checkTry(methodInvocationTree, false);
+        }
+      }
+
+      super.visitMethodInvocation(methodInvocationTree);
+    }
+
+    private void checkTryCatch(MethodInvocationTree methodInvocationTree) {
+      Tree catchParent = ExpressionUtils.getParentOfType(methodInvocationTree, Tree.Kind.CATCH);
+      boolean isError = isCodeInList(methodInvocationTree, errorCodes);
+
+      if (catchParent != null && !isError) {
+        reportIssue(methodInvocationTree, ERROR_ISSUE_MESSAGE);
```

Comment:
Is there any reason to continue with the rest of the method if we report here? Is there a chance we can report here and in a try as well below?
```suggestion
        reportIssue(methodInvocationTree, ERROR_ISSUE_MESSAGE);
        return;
```